libmscore/undo.cpp

```diff
@@ -596,13 +596,30 @@ void Score::undoChangeClef(Staff* ostaff, Segment* seg, ClefType st)
 
 static Element* findLinkedVoiceElement(Element* e, Staff* nstaff)
       {
+      Excerpt* se = e->score()->excerpt();
+      Excerpt* de = nstaff->excerpt();
+      int strack = e->track();
+      int dtrack = nstaff->idx() * VOICES + e->voice();
+
+      if (se)
+            strack = se->tracks().key(strack);
+
+      if (de) {
+            QList<int> l = de->tracks().values(strack);
+            for (int i : l) {
+                  if (nstaff->idx() * VOICES <= i && (nstaff->idx() + 1) * VOICES > i) {
+                        dtrack = i;
```

Comment:
You're right I thought this function just gets called if there is an linked element in the destination. So I'll have to add a new if condition which checks for a null pointer